lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -180,50 +232,63 @@ def add_soft_descriptors(params, options)
       end
 
       def commit(params, options)
-        url = if options[:integration]
-          integration_url
-        elsif test?
-          test_url
-        else
-          live_url
-        end
+        url = base_url(options) + endpoint(params)
 
         if transaction_id = params.delete(:transaction_id)
           url = "#{url}/#{transaction_id}"
         end
 
         begin
-          body = params.to_json
-          response = parse(ssl_post(url, body, headers(body)))
+          response = api_request(url, params)
         rescue ResponseError => e
           response = response_error(e.response.body)
         rescue JSON::ParserError
-          response = json_error(raw_response)
+          response = json_error(e.response.body)
         end
 
         Response.new(
           success_from(response),
           handle_message(response, success_from(response)),
           response,
           test: test?,
-          authorization: authorization_from(params, response),
+          authorization: success_from(response) ? authorization_from(params, response) : 'Authorization Failed',
```

Comment:
Most adapters just let the `Authorization` be `nil` for a failed txn and I think we can let that be the case here too.

Also just FYI "Authorization Failed" wouldn't necessarily be accurate, since the word Authorization could mean the type of action or a response reference which is what it means here.

Reply:
👍